pkg/scheduler/objects/application.go

```diff
@@ -827,13 +827,41 @@ func (sa *Application) tryPlaceholderAllocate(nodeIterator func() NodeIterator,
 			if ph.released || request.taskGroupName != ph.taskGroupName {
 				continue
 			}
+			// before we check anything we need to check the resources equality
+			delta := resources.Sub(ph.AllocatedResource, request.AllocatedResource)
+			// Any negative value in the delta means that at least one of the requested resource in the real
+			// allocation is larger than the placeholder. We need to cancel the placeholder and check the next
+			// placeholder. This should trigger a cleanup of all the placeholder as a task group is always one size.
+			if delta.HasNegativeValue() {
+				log.Logger().Warn("releasing placeholder: real allocation is larger than placeholder",
+					zap.String("requested resource", request.AllocatedResource.String()),
+					zap.String("placeholderID", ph.UUID),
+					zap.String("placeholder resource", ph.AllocatedResource.String()))
+				// release the placeholder and tell the RM
+				ph.released = true
+				sa.notifyRMAllocationReleased(sa.rmID, []*Allocation{ph}, si.TerminationType_TIMEOUT, "cancel placeholder: resource incompatible")
```

Comment:
> cancel placeholder: resource incompatible

Will it be easier to understand if we say "requested resource larger than placeholder" directly? "resource incompatible" may also refer to resource type.

Reply:
Correct because the resource could be incompatible. The real allocation has a reference to resource type GPU with a value of 1, but the placeholder does not which results in GPU becoming negative in the `delta`. The same real allocation could also ask for memory 2GB while the placeholder specifies 4GB. The "incompatible resource" covers these mixed cases better than just the larger than message.
The event generated,  which is what the end user sees, has all the detail needed.